tests/units/Ticket.php

```diff
@@ -1077,4 +1077,174 @@ public function testUpdateFollowup() {
+            (int)$fup->add([
+               'tickets_id'   => $tickets_id,
+               'users_id'     => $uid,
+               'content'      => 'A simple followup'
+            ])
+         )->isGreaterThan(0);
+
+         $this->integer(
+            (int)$fup->fields['timeline_position']
+         )->isEqualTo($user['pos']);
+
+         // TicketTask
+         $task = new \TicketTask();
+         $this->integer(
+            (int)$task->add([
+               'tickets_id'   => $tickets_id,
+               'users_id'     => $uid,
+               'content'      => 'A simple Task'
+            ])
+         )->isGreaterThan(0);
+
+         $this->integer(
+            (int)$task->fields['timeline_position']
+         )->isEqualTo($user['pos']);
+
+         // Document and Document_Item
+         $doc = new \Document();
+         $this->integer(
+            (int)$doc->add([
+               'users_id'     => $uid,
+               'tickets_id'   => $tickets_id,
+               'name'         => 'A simple document object'
+            ])
+         )->isGreaterThan(0);
+
+         $doc_item = new \Document_Item();
+         $this->integer(
+            (int)$doc_item->add([
+               'users_id'      => $uid,
+               'items_id'      => $tickets_id,
+               'itemtype'      => 'Ticket',
+               'documents_id'  => $doc->getID()
+            ])
+         )->isGreaterThan(0);
+
+         $this->integer(
+            (int)$doc_item->fields['timeline_position']
+         )->isEqualTo($user['pos']);
+
+         // TicketValidation
+         $val = new \TicketValidation();
+         $this->integer(
+            (int)$val->add([
+               'tickets_id'   => $tickets_id,
+               'comment_submission'      => 'A simple validation',
+               'users_id_validate' => 5, // normal
+               'status' => 2
+            ])
+         )->isGreaterThan(0);
+
+         $this->integer(
+            (int)$val->fields['timeline_position']
+         )->isEqualTo($user['pos']);
+      }
+   }
+
+   protected function _testGetTimelinePositionSolution($tlp, $tickets_id) {
+      foreach ($tlp as $users_name => $user) {
+         $auth = new \Auth();
+         $this->boolean((boolean)$auth->login($users_name, $user['pass'], true))->isTrue();
+         $uid = getItemByTypeName('User', $users_name, true);
+
+         // Ticket Solution
+         $tkt = new \Ticket();
+         $this->boolean(
+            (boolean)$tkt->update([
+               'id'   => $tickets_id,
+               'solution'      => 'A simple solution from '.$users_name
+            ])
+         )->isEqualto(true);
+
+         $this->integer(
+            (int)$tkt->getTimelinePosition($tickets_id, 'Solution', $uid)
+         )->isEqualTo($user['pos']);
+      }
+   }
+
+   function testGetTimelinePosition() {
+      // login glpi
+      $auth = new \Auth();
+      $this->boolean((boolean)$auth->login('glpi', 'glpi', true))->isTrue();
```

Comment:
``$this->login()``

Reply:
DbTestCase::login doesn't permit login with 'glpi' user, but only with TU_USER

Reply:
for this exact test , no need for glpi i think

Reply:
I wanted to be sure to get all items

Reply:
The test user is superadmin; it should have all required rights ;)

Reply:
then should I use TU_USER or glpi?

Reply:
only ``$this->login`` (So TU_USER)